java-checks/src/main/java/org/sonar/java/checks/ReplaceLambdaByMethodRefCheck.java

```diff
@@ -87,6 +93,21 @@ private void visitLambdaExpression(LambdaExpressionTree tree) {
     return Optional.empty();
   }
 
+  private static Optional<String> getTypeCast(LambdaExpressionTree lambda) {
+    Tree lambdaBody = lambda.body();
+    if (isBlockWithOneStatement(lambdaBody)) {
+      return getTypeCastFromReturn(((BlockTree) lambdaBody).body().get(0), lambda);
+    }
+    return getTypeCast(lambdaBody, lambda);
+  }
+
+  private static Optional<String> getTypeCastFromReturn(Tree statement, LambdaExpressionTree lambda) {
```

Comment:
The only reason of keeping the lambda as 2nd parameter is to ultimately check for the parameter symbol (3 calls below). Wouldn't it be possible to check this earlier (only one param for instance) and only pass a symbol?